### PR TITLE
Enhance new StatementListTest test

### PR DESCRIPTION
**Pull request by JeroenDeDauw**
Follow up to https://github.com/wmde/WikibaseDataModel/pull/289

### CODE REVIEW

File: tests/unit/Statement/StatementListTest.php [hunk 1]
```diff
@@ -287,11 +280,40 @@ public function testGivenTraversableWithNonStatements_constructorThrowsException
...
+		);
+	}
+
+	public function testGivenArgumentListWithNonStatement_constructorThrowsException() {
+		$statement0 = new Statement( new Claim( $this->newSnak( 42, 'foo' ) ) );
+		$statement1 = new Statement( new Claim( $this->newSnak( 42, 'bar' ) ) );
+		$statement2 = new Statement( new Claim( $this->newSnak( 42, 'baz' ) ) );
+
+		$this->setExpectedException( 'InvalidArgumentException' );
+		new StatementList( $statement0, $statement1, array(), $statement2 );
```
Comment (thiemowmde) [inline tests/unit/Statement/StatementListTest.php]: I find this test a bit to complicated, `new StatementList( $statement, array() )` would be enough.

File: tests/unit/Statement/StatementListTest.php [hunk 2]
```diff
@@ -287,11 +280,40 @@ public function testGivenTraversableWithNonStatements_constructorThrowsException
 		new StatementList( $claimsObject );
 	}
 
-	public function testGivenNonTraversable_constructorThrowsException() {
+	public function testGivenNonTraversableOrArgList_constructorThrowsException() {
```
Comment (thiemowmde) [inline tests/unit/Statement/StatementListTest.php]: This name is a bit odd. Could mean `!( isTraversable || isArgList )` or `( !isTraversable ) || isArgList`. `GivenNull` would be better.

**Comment by thiemowmde** [comment]
+2, since none of my comments is a reason to block this, but you may want to address them.

File: tests/unit/Statement/StatementListTest.php [hunk 3]
```diff
@@ -287,11 +280,40 @@ public function testGivenTraversableWithNonStatements_constructorThrowsException
...
+	}
+
+	public function testCanConstructWithStatementArgumentList() {
+		$statement0 = new Statement( new Claim( $this->newSnak( 42, 'foo' ) ) );
+		$statement1 = new Statement( new Claim( $this->newSnak( 42, 'bar' ) ) );
+		$statement2 = new Statement( new Claim( $this->newSnak( 42, 'baz' ) ) );
+
+		$this->assertEquals(
+			new StatementList( array( $statement0, $statement1, $statement2 ) ),
+			new StatementList( $statement0, $statement1, $statement2 )
```
Comment (thiemowmde) [inline tests/unit/Statement/StatementListTest.php]: Same here, 2 statements would be enough.

File: tests/unit/Statement/StatementListTest.php (hunk 1 above)
Comment (JeroenDeDauw) [inline tests/unit/Statement/StatementListTest.php]: Probably yes

**Pull request merged**